wolfssl/wolfcrypt/cmac.h

```diff
@@ -40,16 +43,23 @@
 /* avoid redefinition of structs */
 #if !defined(HAVE_FIPS) || FIPS_VERSION3_GE(2,0,0)
 
+typedef enum CmacType {
+    WC_CMAC_AES = 1
+} CmacType;
+
 #ifndef WC_CMAC_TYPE_DEFINED
     typedef struct Cmac Cmac;
     #define WC_CMAC_TYPE_DEFINED
 #endif
 struct Cmac {
+    CmacType type;
```

Comment:
Enum types are hard to predict size for and this might break build-time compatibility for some folks. I would recommend adding the type lower down in the structure.

Reply:
Concur.  Moved down.